### PR TITLE
Fix small inconsistency in output dimension of "_get_t5_prompt_embeds" function in sd3 pipeline

### PR DESCRIPTION
**Pull request by alirezafarashah**
# What does this PR do?

This PR fixes a small inconsistency in the output dimension of the `_get_t5_prompt_embeds` function in the Stable Diffusion 3  pipeline.  

Previously, when `self.text_encoder_3` was `None`, the function returned a tensor (torch.zeros) with a sequence length of `self.tokenizer_max_length` (77), which corresponds to the CLIP encoder. However, the T5 text encoder used in SD3 has a different maximum sequence length (256).  

As a result, when `text_encoder_3` was available, the prompt embeddings had a sequence length of 333 (256 from T5 + 77 from CLIP), but when it was not available, the returned tensor had only 154 (77 + 77), leading to an inconsistency in output dimensions in `encode_prompt`.  

### Motivation and Context
This change ensures consistent tensor shapes across different encoder availability conditions in the SD3 pipeline.  
It prevents dimension mismatches and potential runtime errors when `text_encoder_3` is `None`.  

Previously, the zeros tensor used `self.tokenizer_max_length`, which corresponds to CLIP, instead of T5’s longer sequence length.  
This mismatch led to inconsistent embedding dimensions when combining outputs from CLIP and T5 in `encode_prompt`.  

### Changes Made
- Replaced `self.tokenizer_max_length` with `max_sequence_length` when returning the zero tensor in `_get_t5_prompt_embeds`, ensuring consistent output dimensions whether `text_encoder_3` is `None` or available.  
  The same `max_sequence_length` parameter is already used in the tokenization step of the same function:
  ```python
  text_inputs = self.tokenizer_3(
      prompt,
      padding="max_length",
      max_length=max_sequence_length,
      truncation=True,
      add_special_tokens=True,
      return_tensors="pt",
  )
- No changes to functionality, inputs, or outputs beyond dimension consistency.



## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

- @asomoza (pipelines and callbacks)  
- @yiyixuxu (pipelines and callbacks)  
- @sayakpaul (general functionalities)

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @.

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Core library:

- Schedulers: @yiyixuxu
- Pipelines and pipeline callbacks: @yiyixuxu and @asomoza
- Training examples: @sayakpaul
- Docs: @stevhliu and @sayakpaul
- JAX and MPS: @pcuenca
- Audio: @sanchit-gandhi
- General functionalities: @sayakpaul @yiyixuxu @DN6

Integrations:

- deepspeed: HF Trainer/Accelerate: @SunMarc
- PEFT: @sayakpaul @BenjaminBossan

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- transformers: [different repo](https://github.com/huggingface/transformers)
- safetensors: [different repo](https://github.com/huggingface/safetensors)

-->

### CODE REVIEW

**Comment by yiyixuxu** [comment]
@bot /style

**Comment by yiyixuxu** [comment]
hey @alirezafarashah 
can you run `make fix-copies"?

**Comment by alirezafarashah** [comment]
> hey @alirezafarashah can you run `make fix-copies"?

Hey @yiyixuxu 
I did and pushed it.
